src/app/gene-browser/gene.ts

```diff
@@ -34,6 +34,25 @@ function collapseTranscripts(transcripts: Transcript[]): Transcript {
   );
 }
 
+function collapseMultipleTranscripts(chromKeys: string[], transcripts: Transcript[]): Transcript[] {
+  let trans: Transcript[] = [];
+  for (const key of chromKeys) {
+    const chromosomeTranscripts = transcripts.filter(t => t.chromosome === key);
+    const allExons: Segment[] = [];
+    const allCodingSequences: Segment[] = [];
+    for (const transcript of chromosomeTranscripts) {
```

Comment:
Try to extract the repeated code from this function and `collapseSegments` into a small helper func